packages/tests/integration/src/relations/ManyToManyCollection.test.ts

```diff
@@ -565,4 +565,19 @@ describe("ManyToManyCollection", () => {
     expect((user as any).comments).not.toBeDefined();
     expect(user.likedComments).toBeDefined();
   });
+
+  it("can identify changes on the many-to-many", async () => {
+    const em = newEntityManager();
+    const book = newBook(em, { title: "To be changed by hook" });
+    const t1 = newTag(em, { name: "t1" });
+    await em.flush();
```

Comment:
Things will sometimes only work because entities are already loaded in memory; instead of using factories to create the book and tag, and can you use the `insert...` methods, and then have the test just do an em.load for both entities (but no populates of the `book.tags` or `tag.books` collections) and watch the same logic work?

Also we should have a test that for m2m removals. I.e. use the same `insert...` methods to start out with an m2m row between a book & tag, and then remove it.

Reply:
I got it, i just added couple of unit test to cover those scenarios, i was missing the removal trigger xd but is working now